reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -181,13 +193,13 @@ public static Subscription emptySubscription() {
 	 * @param e the actual error
 	 */
 	public static void error(Subscriber<?> s, Throwable e) {
-		s.onSubscribe(EmptySubscription.INSTANCE);
+		s.onSubscribe(OnErrorSubscription.INSTANCE);
```

Comment:
@simonbasle after thinking more about it, I ended up with `OnErrorSubscription` in both `#error` and `#reportThrowInSubscribe`.
The meaning of it is the same - we pass a dummy subscription to satisfy spec's 1.9 rule.

But now `#error` and `#reportThrowInSubscribe` look almost identical (except the spec violating catch).
WDYT about adding `try {} catch {}` to `error` and removing `reportThrowInSubscribe`?

Reply:
I'm not comfortable with this modification, as `#error` can be used as part of the normal path. the only situation where the double subscribe _can_ be ignored by StepVerifier is `reportThrowInSubscribe`

Reply:
pushed a new attempt :D